server/src/main/java/org/elasticsearch/cluster/coordination/JoinValidationService.java

```diff
@@ -349,7 +349,7 @@ private ReleasableBytesReference serializeClusterState(DiscoveryNode discoveryNo
                     CompressorFactory.COMPRESSOR.threadLocalOutputStream(Streams.flushOnCloseStream(bytesStream))
                 )
             ) {
-                stream.setVersion(version);
+                stream.setTransportVersion(version.transportVersion);
```

Comment:
This serialised representation of the cluster state needs to go into a map keyed by `TransportVersion`.

Reply:
Good spot